### DISCUSSION TITLE
Materials: Angular or Standard Frequency?

### DISCUSSION THREAD

**Discussion by andrewsalij** [NanoComp/meep#1986] [Q&A] [answered]
There's some contradiction in the Meep documentation regarding initializing materials with dispersion, so I thought I'd ask to clarify things.

From the documentation on LorentzianSusceptibility (https://meep.readthedocs.io/en/latest/Python_User_Interface/#lorentziansusceptibility), it is stated that one needs frequency f_n = \omega_n/2\pi where \omega_n is the angular frequency, and the same for damping.

However, the materials  at https://github.com/NanoComp/meep/blob/master/python/materials.py appears, at least in the cases that I've checked, to use angular frequency. Consider glass, for which
SiO2_frq1 = 1/(0.103320160833333*um_scale)
SiO2_gam1 = 1/(12.3984193000000*um_scale), corresponding to the document http://www.horiba.com/fileadmin/uploads/Scientific/Downloads/OpticalSchool_CN/TN/ellipsometer/Lorentz_Dispersion_Model.pdf, wherein the Lorentz oscillator model is detailed in angular frequency (in eV, which converts to the wavelengths shown above) as is generally done. 

So, which one is it? Thank you in advance.

**Comment by oskooi** [comment, marked as answer]
The ω<sub>t</sub> and Γ<sub>0</sub> fitting parameters from the Lorentzian model provided in https://www.horiba.com/fileadmin/uploads/Scientific/Downloads/OpticalSchool_CN/TN/ellipsometer/Lorentz_Dispersion_Model.pdf are in units of **eV**. To convert eV into μm (for the wavelength), we need to divide the Planck constant [hc](https://en.wikipedia.org/wiki/Planck_constant) = [1.2398419](https://github.com/NanoComp/meep/blob/master/python/materials.py#L11) eV·μm by these quantities. For example, SiO<sub>2</sub> has ω<sub>t</sub> = 12.0 eV = 1.2398419/12.0 μm = 0.10332016 μm. Also, Γ<sub>0</sub> = 0.1 eV = 1.2398419/0.1 μm = 12.3984193 μm.  These are the quantities corresponding to the standard frequency which appear in the materials library:

https://github.com/NanoComp/meep/blob/33cfca1716933512e0999ced187576cfd259182b/python/materials.py#L850-L863

(Also note: the Lorentzian oscillator strength is σ = ε<sub>s</sub> - ε<sub>∞</sub> = 2.12 - 1.0 = 1.12.)

As long as ω and Γ have the *same* units, it does not matter whether they are angular or standard frequency because the units end up cancelling in the equation for the Lorenztian susceptibility.

Note that you can always inspect the complex refractive index using the `epsilon(f)` function of the `Medium` object to verify that the values are what you expect.

```
from meep.materials import SiO2
import numpy as np
import matplotlib
matplotlib.use('agg')
import matplotlib.pyplot as plt

wvl_min = 0.4 # units of μm                                                                                       
wvl_max = 0.7 # units of μm                                                                                       
nwvls = 21

wvls = np.linspace(wvl_min, wvl_max, nwvls)

SiO2_epsilon = np.array([SiO2.epsilon(1/w)[0][0] for w in wvls])

plt.subplot(1,2,1)
plt.plot(wvls,np.real(SiO2_epsilon),'bo-')
plt.xlabel('wavelength (μm)')
plt.ylabel('real(ε)')

plt.subplot(1,2,2)
plt.plot(wvls,np.imag(SiO2_epsilon),'ro-')
plt.xlabel('wavelength (μm)')
plt.ylabel('imag(ε)')

plt.suptitle('SiO$_2$ from Meep materials library')
plt.subplots_adjust(wspace=0.4)
plt.savefig('SiO2_epsilon.png',bbox_inches='tight',dpi=150)
```
![SiO2_epsilon](https://user-images.githubusercontent.com/7152530/157340744-ca0bd795-1f14-41ac-b670-eba2d8ebc936.png)

**Reply by oskooi** [reply]
edit: to convert eV to wavelengths in units of μm, you _always_ need to divide hc (and not ħc) by eV. Thus, the parameters in the materials library are correct.

~~Actually, you raise an important point: if ω<sub>t</sub> and Γ<sub>0</sub> from the reference Horiba notes actually do correspond to the *angular* frequency, then we should be using the Planck's constant of ħc = 0.197326 eV·μm rather than hc=1.2398419 eV·μm to convert these quantities to the vacuum wavelength in units of μm since these quantities are then converted to units of standard frequency for the `frequency` and `gamma` parameters of the `LorentzianSusceptibility` object.~~

**Reply by andrewsalij** [reply]
Thank you for clarifying. Working generally in QM, I had been treating the energy and angular frequency as being the essentially the same (that noted, the conversion to the materials document was reproduced correctly using a script written a while ago that uses the correct Planck's constant). These micron-based units are throwing me a bit for a loop, but I'm sure that they'll become second nature in no time. 

That said, I look forward to continuing to learn and use meep!
